components/shift_reg/shift_reg.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2022 Jaime Albuquerque <jaime.albq@gmail.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * @file shift_reg.h
+ * @defgroup shift_reg shift_reg
+ * @{
+ * 
+ * ESP-IDF driver for generic shift register using 3 pins:
+ * - data
+ * - clock
+ * - latch
+ *
+ */
+
+#if !defined(__SHIFT_REG__H__)
+#define __SHIFT_REG__H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <esp_err.h>
+#include <esp_log.h>
+#include <driver/gpio.h>
+#include <ets_sys.h>
+
+#define SETPIN(PIN)     gpio_set_level(PIN, 1)
+#define CLRPIN(PIN)     gpio_set_level(PIN, 0)
+#define _DELAY_US(x)    ets_delay_us(x)
+
```

Comment:
Since it was ported from a generic library for the IC 74HC595 I just left there. But as this is a specific library for Espressif's MCUs, I'm going to change directly in the code.